great_expectations/expectations/metrics/util.py

```diff
@@ -902,16 +902,26 @@ def get_dialect_like_pattern_expression(  # noqa: C901, PLR0912, PLR0915
         pass
 
     if hasattr(dialect, "dialect"):
-        if issubclass(
-            dialect.dialect,
-            (
-                sa.dialects.sqlite.dialect,
-                sa.dialects.postgresql.dialect,
-                sa.dialects.mysql.dialect,
-                sa.dialects.mssql.dialect,
-            ),
-        ):
-            dialect_supported = True
+        try:
```

Comment:
It doesn't matter, and this isn't on you, but this is just reads so badly / isn't DRY when it feels like it should be. Trying to think if there's a way to dry this up. The only thing that comes to mind is to have a list of functions that get the dialect, something like
```
dialect_getters = [lambda: sa.dialects.sqlite.dialect, lambda: a.dialects.mysql.dialect, ...]
for get_dialect in dialect_getters:
  try:
    if issubclass(dialect.dialect, get_dialect()):
      ..
   except:
   ..
```

Reply:
😂 i did think about this for approximately 5 seconds, but IMO the repetition is way more legible and clear than any shorter option i could think of, outside of rewriting the whole way this function is being used. In any case, not going to make that change here.

Reply:
Cool, works for me!